### ISSUE TITLE
switching to new tab doesn't work when sb.undetectable is set to True

### ISSUE THREAD

**Issue by stefan9999991** [seleniumbase/SeleniumBase#1735]
## Expected Behavior
When using any of the following commands:
`self.switch_to_tab(1)`
`self.switch_to_tab(1, 100)`
`self.switch_to_newest_tab()`
`self.switch_to_window(1)`
`self.switch_to_window(1, 100)`
`self.switch_to_newest_window()`

The program should switch to the newest tab and parse further information in this tab. as mentioned in [the documentation](https://seleniumbase.io/) under Switching Tabs

## Current Behavior
When a new tab is opened, the chrome window does switch to the new tab, but as soon as any operations are made on it, it instantly switches to the old tab and uses the elements found there. Using any of the previous mentioned commands does not change anything about this behavior.

This only happens when `sb.undetectable` is set to `True`. When left to the default value (`False`), the function behaves as expected

## Steps to Reproduce
```
def tab_switch_test(self, url):
        self.open("https://the-internet.herokuapp.com/windows")
        
        click_button = (
	        "div.example"
	        " a"
        )
        
        self.click(click_button)
        # self.switch_to_tab(1, 100)
        self.switch_to_newest_window()
        
        window_test = (
	        "div.example"
	        " h3"
        )
        
        self.assert_element(window_test)
        print(self.find_element(window_test).text)
        
        self.wait(3)
```

The used code should print `New Window` which is the text on the new tab, but the used code actually prints `Opening a new window` which is the text on the old tab.

**Comment by stefan9999991** [comment]
I found a solution to this problem.

Using `self.switch_to_window(0)` the tabs switches to the first tab, but then somehow switches to the second tab.

Even though this works, it is different then

**Comment by stefan9999991** [comment]
Accidentally closed the issue.

**Comment by stefan9999991** [comment]
found a solution (more like a bug which actually does what I want) to this problem.

Using `self.switch_to_window(0)` the tabs switches to the first tab, but somehow after that quickly switches to the second tab.

Even though this works, it is different then what the documentation says for `switch_to_window`:

`Switches control of the browser to the specified window. The window can be an integer: 0 -> 1st tab, 1 -> 2nd tab, etc...`

**Comment by mdmintz** [comment]
I found the issue: https://github.com/ultrafunkamsterdam/undetected-chromedriver/search?q=tab_new is used when clicking on a link. If that opens a new tab, the tab order gets messed up. I'll fix it in the next release.

**Comment by mdmintz** [comment]
Also using https://github.com/ultrafunkamsterdam/undetected-chromedriver/search?q=click_safe for link clicks caused the issue.

**Comment by mdmintz** [comment]
This fix was shipped in ``4.12.6`` - https://github.com/seleniumbase/SeleniumBase/releases/tag/v4.12.6

**Comment by qomhmd** [comment]
I think the problem is back with this settings on Debian 12, using `sbase v4.16.2`:
`uc=True, incognito=True, test=True, headed=True, maximize=True`
